docs/content/guides/columns/column-filter.md

```diff
@@ -155,765 +450,2860 @@ import 'handsontable/dist/handsontable.full.min.css';
+          type: 'numeric',
+          data: 'price',
+          numericFormat: {
+            pattern: '$ 0,0.00',
+            culture: 'en-US',
+          },
+          className: 'htLeft',
+        },
+        {
+          title: 'Date',
+          type: 'date',
+          data: 'sellDate',
+          dateFormat: 'MMM D, YYYY',
+          correctFormat: true,
+          className: 'htRight',
+        },
+        {
+          title: 'Time',
+          type: 'time',
+          data: 'sellTime',
+          timeFormat: 'hh:mm A',
+          correctFormat: true,
+          className: 'htRight',
+        },
+        {
+          title: 'In stock',
+          type: 'checkbox',
+          data: 'inStock',
+          className: 'htCenter',
+        },
+      ]}
+      // enable the column menu
+      dropdownMenu={true}
+      // enable filtering
+      filters={true}
+      // `afterFilter()` is a Handsontable hook: it's fired after each filtering
+      afterFilter={exclude}
+      height="auto"
+      licenseKey="non-commercial-and-evaluation"
+    />
+  );
+};
+
+/* start:skip-in-preview */
+ReactDOM.render(<App />, document.getElementById('exampleExcludeRows'));
+/* end:skip-in-preview */
+```
 
-    } else {
-      setSelectedRowEntries([]);
-    }
+:::
+
+:::
+
+## Use filtering hooks
+
+You can run your code before or after filtering, using the following
+[Handsontable hooks](@/guides/getting-started/events-and-hooks.md):
+
+- [`beforeFilter()`](@/api/hooks.md#beforefilter)
+- [`afterFilter()`](@/api/hooks.md#afterfilter)
+
+For example, you can use [`beforeFilter()`](@/api/hooks.md#beforefilter) for server-side filtering,
+or use [`afterFilter()`](@/api/hooks.md#afterfilter) to
+[exclude rows from filtering](#exclude-rows-from-filtering).
+
+::: only-for javascript
+
+```js
+const configurationOptions = {
+  beforeFilter() {
+    // add your code here
+    return false; // to block front-end filtering
+  },
+  afterFilter() {
+    // add your code here
+  },
+};
+```
+
+:::
+
+::: only-for react
+
+```jsx
+<HotTable
+  beforeFilter={
+    // add your code here
+    return false; // to block front-end filtering
+  }
+  afterFilter={
+    // add your code here
   }
+/>
+```
 
-  let updateSelectedColumn;
-  let applyFilter;
-  let clearFilter;
+:::
 
-  useEffect(() => {
-    const hot = hotRef.current.hotInstance;
-    const filtersPlugin = hot.getPlugin('filters');
+## Control filtering programmatically
 
-    if (colHeaders.length === 0) {
-      const fetchedEntries = hot.getSourceDataAtCol(0);
+You can control filtering at the grid's runtime by using Handsontable's
+[hooks](@/guides/getting-started/events-and-hooks.md) and [API methods](@/api/filters.md#methods).
+This allows you to:
 
-      setColHeaders(hot.getColHeader());
-      setRowEntries(fetchedEntries);
-      setSelectedRowEntries(fetchedEntries);
-    }
+- Enable or disable filtering based on specified conditions. For example, you can disable filtering
```

Comment:
I'm not sure I understand I would I want to do disable filtering for a large data set.